Products/Rider.md

```diff
@@ -0,0 +1,213 @@
+
+In all other cases, we would need to re-implement missing parts using the [IntelliJ Platform SDK](http://www.jetbrains.org/intellij/sdk/docs/welcome.html), which is structured similar to this guide. For interprocess communication, we can easily [extend the protocol]() using a Kotlin DSL.
+
+## Plugin project (.NET)
+
+The standard approach of migrating an existing ReSharper plugin to also work for Rider, is to copy the existing CSPROJ file and adapt the package reference from `JetBrains.ReSharper.SDK` to `JetBrains.Rider.SDK`.
+
+> **NOTE:** It is strongly recommended to use SDK-based project files to decrease friction.
+
+
+Those two project files could be named like `MyPlugin.RS.csproj` and `MyPlugin.RD.csproj`, which is great for code sharing. Unfortunately, this won't play well when executing MSBuild on them, because both projects would share the same `project.assets.json` file. A simple solution to fix this is to put a custom `Directory.Builds.props` file besides the two project files:
+
+```xml
+<Project>
+  <PropertyGroup>
+    <RootNamespace>MyPlugin</RootNamespace>
+    <DefaultItemExcludes>$(DefaultItemExcludes);obj\**</DefaultItemExcludes>
+    <BaseIntermediateOutputPath>obj\$(MSBuildProjectName)\</BaseIntermediateOutputPath>
+    <OutputPath>bin\$(MSBuildProjectName)\$(Configuration)\</OutputPath>
+  </PropertyGroup>
+</Project>
+```
+
+This will provide MSBuild with an individual OBJ and BIN folder both - the ReSharper and Rider project. 
+
+## Plugin packaging
+
+Rider plugins are simple ZIP archives containing metadata about the plugin, ReSharper extensions (DLL) and/or IntelliJ extensions (JAR). They can be installed the IntelliJ way - through `Settings | Plugins` (tip: during development, we can chose to *install plugins from disk*).
+
+> **NOTE:** Archives created using .NET capabilities (like `System.IO.Compression` or PowerShell's `Compress-Archive`) [might not work](https://youtrack.jetbrains.com/issue/IDEA-180829).
+
+The content should be structured like this:
+
+```
++ META-INF
+   - plugin.xml
++ lib
+   - plugin.jar
++ dotnet
+   - plugin.dll
+```
+
+In case we have an IntelliJ part in our plugin, the `plugin.xml` usually gets compiled into the JAR file. It should contain at least the following metadata to be installed from Rider:
+
+```xml
+<idea-plugin>
+  <depends>com.intellij.modules.rider</depends>
+  <idea-version since-build="181.4379" until-build="182.*"/>
+  <extensions defaultExtensionNs="com.company.plugin" />
+  
+  <id>com.company.plugin</id>
+  <version>1.3.3.7</version>
+  <name>MyCompanyPlugin</name>
+  <vendor url="https://www.company.com">MyCompany</vendor>
+  <description>Some description.</description>
+  <change-notes>Some change-notes.</change-notes>
+</idea-plugin>
+```
+
+Currently, the `idea-version` must be updated manually. The value `since-build` can be found in the *About JetBrains Rider* dialog, while the `until-build` is usually just `<next-major>.*`.
```

Comment:
Do we have docs on the major version naming? (year + release version . something), may be good to link here.